File: src/Entity/Dermatologue.php
<?php

namespace App\Entity;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use App\Repository\DermatologueRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Table(name="dermatologue")
 * @ORM\Entity(repositoryClass=DermatologueRepository::class)
 *
 *
 */
class Dermatologue extends User
{


   /*
    /**
     *
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="Please enter your identifier")
     * @Assert\Length(
     *     min=8,
     *
     *    maxMessage = "Your cin cannot be longer than {{ limit }} characters"
     * )
     * @@Assert\Positive
     */
/*    private $cin;*/
    /**
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     */
    private $id;
    /**
     *
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please enter your diploma")
     */
    private $diplome;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please enter your formation")
     */
    private $formation;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please make a choice")
     */
    private $langue;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please make a choice")
     */
    private $horaire;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please make a choice")
     */
    private $modereglement ;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please make a choice")
     */
    private $assurancemaladie;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Please upload image")
     * @Assert\File(mimeTypes={"image/jpeg"})
     */
    private $image;

    /**
     * @ORM\ManyToOne(targetEntity=CategorieDerm::class, inversedBy="dermatologues" )
     * @ORM\JoinColumn(nullable=false ,name="categorie", referencedColumnName="ref")
     */
    private $categorie;




    public function getId(): ?int
    {
        return parent::getId();
    }

        public function getDiplome(): ?string
    {
        return $this->diplome;
    }

    public function setDiplome(string $diplome): self
    {
        $this->diplome = $diplome;

        return $this;
    }

    public function getFormation(): ?string
    {
        return $this->formation;
    }

    public function setFormation(string $formation): self
    {
        $this->formation = $formation;

        return $this;
    }

    public function getLangue(): ?string
    {
        return $this->langue;
    }

    public function setLangue(string $langue): self
    {
        $this->langue = $langue;

        return $this;
    }

    public function getHoraire(): ?string
    {
        return $this->horaire;
    }

    public function setHoraire(string $horaire): self
    {
        $this->horaire = $horaire;

        return $this;
    }

    public function getModereglement()
    {
        return $this->modereglement;
        //$modereglemen[] = 'Cheque';
        //return array_unique($modereglemen);

    }

    public function setModereglement( string $modereglement)
    {
        $this->modereglement  = $modereglement;

        return $this;
    }

    public function getAssurancemaladie(): ?string
    {
        return $this->assurancemaladie;
    }

    public function setAssurancemaladie(string $assurancemaladie): self
    {
        $this->assurancemaladie = $assurancemaladie;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage( $image)
    {
        $this->image = $image;

        return $this;
    }

    public function getCategorie(): ?CategorieDerm
    {
        return $this->categorie;
    }

    public function setCategorie(?CategorieDerm $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }
    /*
    /** @see \Serializable::serialize() */
/*public function serialize()
{
    return serialize(array(
        $this->modereglement,

    ));
}

    /** @see \Serializable::unserialize() */
  /*  public function unserialize($serialized)
    {
        list (
            $this->modereglement,

            // see section on salt below
            // $this->salt
            ) = unserialize($serialized);
    }
*/


}

File: sprint java/Entities/Dermatologue.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

/**
 *
 * @author IMNA
 */
public class Dermatologue extends User {
    int idDerm   ; 
    int categorie ;
    String diplome,formation,langue,horaire,modereglement,assurancemaladie,image;
    
    public Dermatologue() {
    }

    public Dermatologue(int id, String diplome, String formation, String langue, String horaire, String modereglement, String assurancemaladie, String image , int categorie) {
        this.idDerm = id;
        this.diplome = diplome;
        this.formation = formation;
        this.langue = langue;
        this.horaire = horaire;
        this.modereglement = modereglement;
        this.assurancemaladie = assurancemaladie;
        this.image = image;
        this.categorie=categorie;
        
    }
    

    public int getIdDerm() {
        return idDerm;
    }

    public void setIdDerm(int id) {
        this.idDerm = id;
    }

    public String getDiplome() {
        return diplome;
    }

    public void setDiplome(String diplome) {
        this.diplome = diplome;
    }

    public String getFormation() {
        return formation;
    }

    public void setFormation(String formation) {
        this.formation = formation;
    }

    public String getLangue() {
        return langue;
    }

    public void setLangue(String langue) {
        this.langue = langue;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }

    public String getModereglement() {
        return modereglement;
    }

    public void setModereglement(String modereglement) {
        this.modereglement = modereglement;
    }

    public String getAssurancemaladie() {
        return assurancemaladie;
    }

    public void setAssurancemaladie(String assurancemaladie) {
        this.assurancemaladie = assurancemaladie;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getCategorie() {
        return categorie;
    }

    public void setCategorie(int categorie) {
        this.categorie = categorie;
    }
    

    @Override
    public String toString() {
        return "Dermatologue{" + "id=" + idDerm + ", diplome=" + diplome + ", formation=" + formation + ", langue=" + langue + ", horaire=" + horaire + ", modereglement=" + modereglement + ", assurancemaladie=" + assurancemaladie + ", image=" + image + '}';
    }
    
}
